zooniverse/Panoptes-Front-End
# Decaffeinate CommentBox + improve Talk error messages

## PR Overview

Staging branch URL: https://pr-7161.pfe-preview.zooniverse.org
Closes #7021

This PR rewrites the `<CommentBox>` component to address two goals: (A) add more descriptive error messages, and (B) convert the code from CoffeeScript to JavaScript

- the CommentBox component is used in 1. creating Discussions, 2. adding a new Comment to a Discussion, and 3. sending direct messages to other users
- Dev note: The semi-automated conversion from CoffeeScript (via [decaffeinate](https://decaffeinate-project.org/)) was just par for the course, as it was easier than trying to keep the CoffeeScript code working.

Expected behaviour:
- If you're logged in as a normal, standard, typical, run-of-the-mill user, you should be able to...
  - Create a discussion on a project (e.g. https://www.zooniverse.org/projects/darkeshard/test-project-2022/talk/269?env=staging)
  - Create a comment in a discussion (e.g. https://www.zooniverse.org/projects/darkeshard/test-project-2022/talk/269/407?env=staging)
  - Send direct messages to other volunteers (https://www.zooniverse.org/inbox)
- If you're a user that hasn't confirmed their email yet, then...
  - ...you should not be able to do any of the above,
  - 🆕 ...and the **error message** when you click on the **"Submit"** button should be `"Your account email address is not yet confirmed. Please check your email for confirmation instructions, or visit https://www.zooniverse.org/settings/email to request new confirmation email."`
  - (Previously, this error message would either be `"not allowed to create this Discussion"` or `"not allowed to create this Discussion"` or `You must confirm your account `)
- If you're a user that has been banned, then...
  - ...you should not be able to do any of the above,
  - 🆕 ...and the **error message** when you click on the **"Submit"** button should be `You are banned. Email [contact@zooniverse.org](mailto:contact@zooniverse.org) for further information.`
  - (Previously, this error message would just be `"You are banned"`)
  - ⚠️ MAJOR NOTE: a banned user may actually have trouble even reaching the pages which have the components that implement these error messages, but that's beyond the scope of this PR.

### Testing

Testing steps:
- Login with a standard run-of-the-mill user account with a confirmed email address, e.g. `zootester1`
  - Go to https://pr-7161.pfe-preview.zooniverse.org/projects/darkeshard/test-project-2022/talk/269?env=staging and create a discussion.
  - Go to https://pr-7161.pfe-preview.zooniverse.org/projects/darkeshard/test-project-2022/talk/269/407?env=staging and post a comment 
  - Go to https://pr-7161.pfe-preview.zooniverse.org/inbox?env=staging and send a message (to `darkeshard` or `zootester2` or something)
  - Confirm that behaviour matches expected behaviour.
- Login with a user account with an unconfirmed email address and that's _not_ banned, e.g. `zootester2024b`
  - Repeat all of the above
- Login with any test account, e.g. zootester2024b
  - Repeat all of the above, BUT at _each step,_ right before pressing the "Submit" button, ban the user account (via a separate window where you're logged in as an admin)
  - Then, un-ban the test account before moving to the next step.
  - This is required because _a banned user can't even load the Talk boards to begin with._ 

NOTE: being banned supersedes being unconfirmed, so a banned && unconfirmed user will only receive "you are banned" error messages.

### Status

Ready for review.

I'd appreciate some very thorough tests on posting comments, as this affects Talk.
